City-of-Helsinki/drupal-asuntotuotanto
# SATT-103: Satt-103: fixed field_hero_design related bug and  is deprecated

### Description

Fixed 
`{"message":"Warning: Undefined array key \"value\" in Drupal\\options\\Plugin\\Field\\FieldType\\ListItemBase::simplifyAllowedValues() (line 521 of /app/public/core/modules/options/src/Plugin/Field/FieldType/ListItemBase.php) #0 /app/public/core/includes/bootstrap.inc(166): _drupal_error_handler_real()\n#1 /app/public/core/modules/options/src/Plugin/Field/FieldType/ListItemBase.php(521): _drupal_error_handler()\n#2 /app/public/core/modules/options/src/Plugin/Field/FieldType/ListItemBase.php(496): Drupal\\options\\Plugin\\Field\\FieldType\\ListItemBase::simplifyAllowedValues()`

and 

`{"message":"Deprecated function: Creation of dynamic property Drupal\\asu_application\\Form\\SalespersonApplicationForm::$parseModeManager is deprecated in Drupal\\asu_application\\Form\\SalespersonApplicationForm->__construct() (line 43 of /app/public/modules/custom/asu_application/src/Form/SalespersonApplicationForm.php)`


### How to test

- make fresh
- check that make fresh goes without error
- login as admin
- rebuild track information and re-index search api indexes
- logout
- login as salesperson like uid 85
- go users page and create a application for user "Luo hakemus uudelle käyttäjälle"
- check that you can create application for user without error
